fatal error: 'stdint.h' file not found with Bazel@HEAD

https://buildkite.com/bazel/bazel-at-head-plus-downstream/builds/3400#018b45b4-c2c2-41de-858e-d02253dab2ba

Platform : MacOS

Logs:
```
ERROR: /Users/buildkite/builds/bk-imacpro-20/bazel-downstream-projects/rules_rust/examples/bindgen/BUILD.bazel:10:21: Generating bindings for bindgen/simple.h.. failed: (Exit 101): bindgen-cli failed: error executing RustBindgen command (from target //bindgen:simple_bindgen__bindgen)
bindgen/simple.h:1:10: fatal error: 'stdint.h' file not found
panicked at external/rules_rust_bindgen__bindgen-cli-0.65.1/main.rs:52:36:
Unable to generate bindings: ClangDiagnostic("bindgen/simple.h:1:10: fatal error: 'stdint.h' file not found\n")

```

Steps:
```
git clone https://github.com/bazelbuild/rules_rust.git
git reset 3a78d01631ea8c2f7b747f6f0e7b4d29c735371d --hard
export USE_BAZEL_VERSION=29985cd45fb843338b312fe0b286553b178fc4c9
cd examples
bazel build  //...
```

CC Greenteam @fweikert

--- a/examples/WORKSPACE.bazel
+++ b/examples/WORKSPACE.bazel
@@ -179,3 +179,12 @@ rbe_preconfig(
     name = "buildkite_config",
     toolchain = "ubuntu1804-bazel-java11",
 )
+
+load("@build_bazel_apple_support//crosstool:setup.bzl", "apple_cc_configure")
+
+# As of Bazel 7, Bazel's bundled cc_toolchain only supports using CommandLineTools and not a full Xcode.
+# This manifests itself if you try to use anything which requires access to $DEVELOPER_DIR, such as our bindgen example which non-hermetically tries to read libc++ headers from the system.
+# On CI (and for some of our contributors), we have a full Xcode install, so this fails.
+# We register the apple_support cc_toolchain here because it supports both CommandLineTools and full Xcodes.
+# See https://github.com/bazelbuild/rules_rust/issues/2207 and https://github.com/bazelbuild/bazel/issues/20638
+apple_cc_configure()

